zerver/views/navigation_views.py

```diff
@@ -0,0 +1,108 @@
+from typing import Annotated
+
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import gettext as _
+from pydantic import Json, StringConstraints
+
+from zerver.actions.navigation_views import (
+    do_add_navigation_view,
+    do_remove_navigation_view,
+    do_update_navigation_view,
+)
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.navigation_views import get_navigation_view, get_navigation_views_for_user
+from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import typed_endpoint
+from zerver.models import NavigationView, UserProfile
+
+BUILT_IN_VIEW_FRAGMENT = [
```

Comment:
Let's call this `BUILT_IN_VIEW_FRAGMENTS` -- it should be obvious it's plural.

Also, I think it should be a set, not a list? Doesn't matter much for performance, but that is the right data structure.